### PR TITLE
Fix #5267: Not able to navigate back from story. 

### LINKED ISSUE #5267: [BUG]: Not able to navigate back from story.
### Describe the bug

when user click on view all in **stories for you** and open any story, user will be stuck, there is no way to go back to home.

### Steps To Reproduce

1. Click on view all
2. open any story
3. click on close icon and try to go back.

### Expected Behavior

Clicking back or close button should navigate back.

### Screenshots/Videos

[storyissue.webm](https://github.com/oppia/oppia-android/assets/76042077/b2c3c316-80bd-4aa4-8b0f-ba879706df7d)

### What device/emulator are you using?

_No response_

### Which Android version is your device/emulator running?

_No response_

### Which version of the Oppia Android app are you using?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Vishwajith-Shettigar**
## Explanation

Fixes #5267
Destroyed ExplorationActivity before creating TopicActivity to prevent loop between the same two activity, when user tries to go back.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## Before
 [no_way_home.webm](https://github.com/oppia/oppia-android/assets/76042077/8c2afa69-9237-4f0d-8b53-f1e89851bd4a) 

## After
[home_coming.webm](https://github.com/oppia/oppia-android/assets/76042077/92e3bc79-b042-4610-bed5-def7dda2da2c)

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Vishwajith-Shettigar** [comment]
Hi @adhiamboperes, based on my testing, I haven't encountered any regression errors. Please take a high-level pass.

**Comment by Vishwajith-Shettigar** [comment]
Hi @adhiamboperes, based on my testing, I haven't encountered any regression errors. Please take a low-level pass.
Thank you.

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt [hunk 1]
```diff
@@ -198,7 +198,6 @@ class RecentlyPlayedFragmentPresenter @Inject constructor(
             parentScreen = ExplorationActivityParams.ParentScreen.PARENT_SCREEN_UNSPECIFIED,
             isCheckpointingEnabled = canHavePartialProgressSaved
           )
-          activity.finish()
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt]: Not sure why this needed to b e removed?

**Review by adhiamboperes — COMMENTED** [review]
Thanks @Vishwajith-Shettigar!

Please see my comment in-line.

File: app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt (hunk 1 above)
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt]: I think that change is not needed.

**Comment by Vishwajith-Shettigar** [comment]
@adhiamboperes, PTAL.

**Review by adhiamboperes — APPROVED** [review]
LGTM, Thanks @Vishwajith-Shettigar!

**Pull request merged**